### PR TITLE
Functions - Factorial Iterative

### PR DESCRIPTION
**Pull request by Abduliitr**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [ ] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [x] Author
    - [x] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [x] Author
    - [x] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [x] Author
    - [x] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Comment by Abduliitr** [comment]
The solution is failing for factorial of `100`.

If we want the code to be able to find factorial of large numbers as well, a different logic will be required. I believe this problem is targeted for beginners, so such large test should not be there.

**Comment by sailesh-ramayanam** [comment]
> The solution is failing for factorial of `100`.
> 
> If we want the code to be able to find factorial of large numbers as well, a different logic will be required, which will increase the difficulty level of the problem. I believe this problem is targeted for beginners, so such large test should not be there.

Agree.
@satya-9 Please edit the test cases. Please make sure the result of each test case is less than [MAX_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER)

File: solutions/functions/factorial-iterative.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+
+function factorial(num){
+    if(num < 0) return "undefined";
```
Comment (sailesh-ramayanam) [inline solutions/functions/factorial-iterative.js:15]: @Abduliitr Please use curly braces.

Comment (Abduliitr) [reply solutions/functions/factorial-iterative.js]: Used.

**Comment by satya-9** [comment]
> 


> > The solution is failing for factorial of `100`.
> > If we want the code to be able to find factorial of large numbers as well, a different logic will be required, which will increase the difficulty level of the problem. I believe this problem is targeted for beginners, so such large test should not be there.
> 
> Agree. @satya-9
> 
> 1. Please edit the test cases. Please make sure the result of each test case is less than [MAX_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER)
> 2. Also, please add a template for function



1. > @sailesh-ramayanam I have edited the test cases for an input of 10 to  output of 3628800. Please run problem once.
2. Have added the template for function

**Review by sailesh-ramayanam — APPROVED** [review]

**Review by satya-9 — APPROVED** [review]

**Pull request merged**